### ISSUE TITLE
Disabled IP-in-IP tunneling,but it still creates tunnel,which cause the network trouble

### ISSUE THREAD

**Issue by yinwenqin** [cloudnativelabs/kube-router#728]
Our kubernetes nodes are in two subnet: 192.168.20.0/24 and 192.168.9.0/24(master node in this subnet).Routing Infrastructure Outside Cluster route traffic for pod-to-pod networking across nodes.So i configured --enable-overlay=false to disable IP-in-IP tunneling,it's been working well for months.

  But since last week, problems have arisen, and occasionally several nodes in our 192.168.20.0/24 network segment have automatically configured tunneling routing entries: unilaterally, tunnels have been established for 192.168.9.0/24 subnetworks, while gateway 192.168.9.254 has not established tunnels, which has led to network failures to reach 192.168.9.0/24 subnet.The kubernetes master nodes are in 192.168.9.0/24 subnet,so the problem node becomes into unready status.

   This is a tunnel routing entry created automatically on the problematic node:
```
# 172.26.0.0/16 is the pod cidr,the last routing entry to 192.168.9.0/24 is the fuse of the failure.

[root@yksp020204 ~]# ip route | grep tun
172.26.0.0/24 dev tun-192168928 proto 17 src 192.168.20.204 
172.26.1.0/24 dev tun-192168927 proto 17 src 192.168.20.204 
172.26.2.0/24 dev tun-192168929 proto 17 src 192.168.20.204 
172.26.12.0/24 dev tun-192168930 proto 17 src 192.168.20.204 
192.168.9.0/24 dev tun-1921689254 proto 17 src 192.168.20.204

```
 
 Here's the log before kube-router quit because he couldn't connect to master:
```
[root@yksp020204 ~]# docker ps | grep router
a5352430380c        cloudnativelabs/kube-router                "/usr/local/bin/kube…"   About a minute ago   Up About a minute                       k8s_kube-router_kube-router-dl9h6_kube-system_74318d75-4af8-11e9-8479-3440b5a2bb9c_55
0f651d7d2d29        gcr.io/google_containers/pause-amd64:3.0   "/pause"                 4 weeks ago          Up 4 weeks                              k8s_POD_kube-router-dl9h6_kube-system_74318d75-4af8-11e9-8479-3440b5a2bb9c_3
[root@yksp020204 ~]# docker logs a5352430380c
I0506 00:48:08.168780       1 kube-router.go:207] Running /usr/local/bin/kube-router version v0.3.0, built on 2019-04-29T04:21:18+0000, go1.10.8
I0506 00:48:08.803693       1 network_policy_controller.go:142] Starting network policy controller
I0506 00:48:08.835491       1 network_routes_controller.go:980] Could not find annotation `kube-router.io/bgp-local-addresses` on node object so BGP will listen on node IP: 192.168.20.204 address.
I0506 00:48:08.861401       1 network_services_controller.go:284] Starting network services controller
I0506 00:48:09.026098       1 network_routes_controller.go:223] Starting network route controller
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.9.254" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.9.27" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.9.28" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.9.29" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.9.30" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.40" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.41" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.107" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.109" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.112" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.114" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.141" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.203" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.206" Topic=Peer
time="2019-05-06T00:48:09Z" level=info msg="Add a peer configuration for:192.168.20.113" Topic=Peer
time="2019-05-06T00:48:19Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.9.27 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:19Z" level=info msg="Peer Up" Key=192.168.9.27 State=BGP_FSM_OPENCONFIRM Topic=Peer
time="2019-05-06T00:48:19Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.114 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:19Z" level=info msg="Peer Up" Key=192.168.20.114 State=BGP_FSM_OPENCONFIRM Topic=Peer
I0506 00:48:19.800233       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:19.803017       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:19.841657       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:19.844282       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
time="2019-05-06T00:48:20Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.9.28 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:20Z" level=info msg="Peer Up" Key=192.168.9.28 State=BGP_FSM_OPENCONFIRM Topic=Peer
time="2019-05-06T00:48:20Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.9.29 State=BGP_FSM_OPENSENT Topic=Peer
I0506 00:48:20.484678       1 network_routes_controller.go:419] Cleaning up old routes if there are any
time="2019-05-06T00:48:20Z" level=info msg="Peer Up" Key=192.168.9.29 State=BGP_FSM_OPENCONFIRM Topic=Peer
I0506 00:48:20.487607       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:20.524177       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:20.526058       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
time="2019-05-06T00:48:21Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.141 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:21Z" level=info msg="Peer Up" Key=192.168.20.141 State=BGP_FSM_OPENCONFIRM Topic=Peer
time="2019-05-06T00:48:21Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.206 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:21Z" level=info msg="Peer Up" Key=192.168.20.206 State=BGP_FSM_OPENCONFIRM Topic=Peer
I0506 00:48:21.807274       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:21.809543       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:21.809741       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:21.810908       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
time="2019-05-06T00:48:22Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.40 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:22Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.107 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:22Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.9.30 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:22Z" level=info msg="Peer Up" Key=192.168.20.40 State=BGP_FSM_OPENCONFIRM Topic=Peer
time="2019-05-06T00:48:22Z" level=info msg="Peer Up" Key=192.168.9.30 State=BGP_FSM_OPENCONFIRM Topic=Peer
time="2019-05-06T00:48:22Z" level=info msg="Peer Up" Key=192.168.20.107 State=BGP_FSM_OPENCONFIRM Topic=Peer
I0506 00:48:22.818749       1 network_routes_controller.go:419] Cleaning up old routes if there are any
time="2019-05-06T00:48:22Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.203 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:22Z" level=info msg="Peer Up" Key=192.168.20.203 State=BGP_FSM_OPENCONFIRM Topic=Peer
I0506 00:48:22.820390       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:22.820532       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:22.821464       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:22.854111       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:22.855616       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:22.855767       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:22.857279       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
time="2019-05-06T00:48:23Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.109 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:23Z" level=info msg="Peer Up" Key=192.168.20.109 State=BGP_FSM_OPENCONFIRM Topic=Peer
I0506 00:48:23.803003       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:23.804698       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
time="2019-05-06T00:48:24Z" level=info msg="Can't find configuration for a new passive connection from:192.168.20.254" Topic=Peer
time="2019-05-06T00:48:25Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.41 State=BGP_FSM_OPENSENT Topic=Peer
time="2019-05-06T00:48:25Z" level=info msg="Peer Up" Key=192.168.20.41 State=BGP_FSM_OPENCONFIRM Topic=Peer
time="2019-05-06T00:48:25Z" level=info msg="skipped asn negotiation: peer-as: 64512, peer-type: internal" Key=192.168.20.113 State=BGP_FSM_OPENSENT Topic=Peer
I0506 00:48:25.801623       1 network_routes_controller.go:419] Cleaning up old routes if there are any
time="2019-05-06T00:48:25Z" level=info msg="Peer Up" Key=192.168.20.113 State=BGP_FSM_OPENCONFIRM Topic=Peer
I0506 00:48:25.804545       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:25.810508       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:25.812974       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
time="2019-05-06T00:48:27Z" level=info msg="Peer Up" Key=192.168.9.254 State=BGP_FSM_OPENCONFIRM Topic=Peer
I0506 00:48:27.135745       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:27.138354       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:27.174845       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:27.177163       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:27.217268       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:27.219504       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node
I0506 00:48:27.260813       1 network_routes_controller.go:419] Cleaning up old routes if there are any
I0506 00:48:27.262794       1 network_routes_controller.go:433] Cleaning up if there is any existing tunnel interface for the node

```

I'm very sure that the overlay option has been turned off. This is my daemonSet configuration option:
```
      - args:
        - --run-router=true
        - --enable-overlay=false
        - --run-firewall=true
        - --run-service-proxy=true
        - --advertise-cluster-ip=true
        - --advertise-external-ip=true
        - --cluster-asn=64512
        - --peer-router-ips=192.168.9.254
        - --peer-router-asns=64513
        - --peer-router-multihop-ttl=255

```

And here's the process inside the kube-router container:

```
root@yksp020204:~ #ps -ef | more
PID   USER     TIME   COMMAND
    1 root       1:51 /usr/local/bin/kube-router --run-router=true --enable-overlay=false --run-firewall=true --run-service-proxy=true --advertise-cluster-ip=true --advertise-external-ip=true --cluster-asn=64512 --peer-router-ips=192.168.9.254 --peer-router-asns=64513 --peer-router-multihop-ttl=255
94950 root       0:00 bash
94995 root       0:00 ps -ef
94996 root       0:00 more

```

Here is the system and software version:
```

[root@yksp020204 ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2017-12-15T21:07:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}

[root@yksp020204 ~]# cat /etc/redhat-release 
CentOS Linux release 7.6.1810 (Core) 

[root@yksp020204 ~]# uname -r
3.10.0-957.10.1.el7.x86_64

[root@yksp020204 ~]# docker images | grep router
cloudnativelabs/kube-router                            latest              aedbcb8038f1        6 days ago          105MB

```

    I guess this is the problem caused by the new version of the mirror, because I saw that the latest mirror was updated six days ago. Can this be confirmed?
   Now every time this problem arises, my temporary solution is to delete these tunnel routing entries, and delete the pod of kube-router to recreate it. I hope to get a permanent and effective solution. Thank you!

**Comment by aauren** [comment]
I believe that this is related to the PR I submitted here: #722

**Comment by ross-w** [comment]
I am also seeing this issue (default route gets replaced with an ipip tunnel despite `--enable-overlay=false`.) Occurred on our development (thankfully) kubernetes cluster after a reboot upgraded kube-router to `v0.3.0`. Forcing version to `v0.2.5` works around the issue for now, but I'll watch the linked PR.

**Comment by MarkDeckert** [comment]
Temporary resolution in v0.3.0 is setting _both_ --enable-overlay=false --overlay-type=subnet

**Comment by murali-reddy** [comment]
Please checkout 0.3.1 release which addresses this issue.

**Comment by kunalpuriii** [comment]
@murali-reddy i have not mentioned - --enable-overlay=false in the args as default is true, but i dont see any tunnel neither in logs or in routing table. All i can see bgp establishment.

Please help
--- BGP Server Configuration ---
AS:        64512
Router-ID: 192.168.5.22
Listening Port: 179, Addresses: 192.168.5.22, ::1

--- BGP Neighbors ---
Peer            AS  Up/Down State       |#Received  Accepted
192.168.5.11 64512 00:26:30 Establ      |        1         1
192.168.5.12     *    never Active      |        0         0
192.168.5.21 64512 00:26:27 Establ      |        1         1

--- BGP Route Info ---
   Network              Next Hop             AS_PATH              Age        Attrs
*> 10.1.0.0/24          192.168.5.11                              00:26:30   [{Origin: i} {LocalPref: 100}]
*> 10.1.1.0/24          192.168.5.21                              00:26:27   [{Origin: i} {LocalPref: 100}]
*> 10.1.2.0/24          192.168.5.22                              00:03:42   [{Origin: i}]

--- IPVS Services ---
IP Virtual Server version 1.2.1 (size=4096)
Prot LocalAddress:Port Scheduler Flags
  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn

**Comment by aauren** [comment]
Fixed via #722 

As to the last comment, does not relate to this issue. Feel free to open a separate issue if you're still being affected.
